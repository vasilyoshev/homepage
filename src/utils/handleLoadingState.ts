import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
