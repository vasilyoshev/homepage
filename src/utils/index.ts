export * from './handleLoadingState';
export * from './testUtils';
