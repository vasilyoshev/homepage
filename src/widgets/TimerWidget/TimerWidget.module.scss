.timers {
  width: 180px;
  margin: 0 auto;
  text-align: center;
}

.display {
  margin-bottom: 20px;

  span {
    font-size: 36px;
  }
}
