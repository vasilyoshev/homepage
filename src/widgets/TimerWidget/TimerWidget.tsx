import React, { useState, useEffect } from 'react';
import styles from './TimerWidget.module.scss';
import { TimerWidgetControls } from 'components';

export const TimerWidget: React.FC = () => {
  const [time, setTime] = useState(0);
  const [timerOn, setTimerOn] = useState(false);

  useEffect(() => {
    let interval = 0;
    if (timerOn) {
      interval = window.setInterval(() => {
        setTime((prevTime) => prevTime + 10);
      }, 10);
    } else if (!timerOn) {
      clearInterval(interval);
    }
    return () => clearInterval(interval);
  }, [timerOn]);

  const startTimer = () => setTimerOn(true);
  const stopTimer = () => setTimerOn(false);
  const resetTimer = () => setTime(0);
  const resumeTimer = () => setTimerOn(true);

  return (
    <main className={styles.timers}>
      <h2>Timer</h2>
      <section className={styles.display}>
        <span>{('0' + Math.floor((time / 60000) % 60)).slice(-2)}:</span>
        <span>{('0' + Math.floor((time / 1000) % 60)).slice(-2)}:</span>
        <span>{('0' + ((time / 10) % 100)).slice(-2)}</span>
      </section>
      <TimerWidgetControls
        time={time}
        timerOn={timerOn}
        startTimer={startTimer}
        stopTimer={stopTimer}
        resetTimer={resetTimer}
        resumeTimer={resumeTimer} />
    </main>
  );
};
