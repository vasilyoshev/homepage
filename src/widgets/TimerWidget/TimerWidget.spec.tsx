import { render } from 'utils';
import { TimerWidget } from './TimerWidget';

describe('TimerWidget', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<TimerWidget />);

    expect(asFragment()).toMatchSnapshot();
  });
});
