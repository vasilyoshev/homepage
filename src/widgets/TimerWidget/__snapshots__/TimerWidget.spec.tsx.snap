// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerWidget renders successfully 1`] = `
<DocumentFragment>
  <main
    class="timers"
  >
    <h2>
      Timer
    </h2>
    <section
      class="display"
    >
      <span>
        00:
      </span>
      <span>
        00:
      </span>
      <span>
        00
      </span>
    </section>
    <section
      class="buttons"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Start
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </section>
  </main>
</DocumentFragment>
`;
